integrationTests/testWebServer.go

```diff
@@ -144,3 +147,23 @@ func loadResponse(t *testing.T, rsp io.Reader, destination interface{}) {
 
 	assert.Nil(t, err)
 }
+
+// WaitTimeout returns true if work group waiting timed out
+func WaitTimeout(t *testing.T, wg *sync.WaitGroup, timeout time.Duration) bool {
```

Comment:
Would this version work too?
```go
func WaitTimeout(wg *sync.WaitGroup, timeout time.Duration) bool {
	done := make(chan struct{})

	go func() {
		defer close(done)
		wg.Wait()
	}()

	select {
	case <-done:
		return false
	case <-time.After(timeout):
		return true
	}
}
```

If not, at least we could remove `t *testing.T`, or even better: make the test fail inside this func, instead of failing it on caller side

Reply:
yes, refactored to use this version :+1: